### PR TITLE
[Fix #3936] Add new `Style/MixinGrouping` cop

### LINKED ISSUE #3936: Cop idea: enforce a style for `include`, `extend`, `prepend`, etc.
I see two distinct styles for mixing in modules:

### Style 1

```ruby
class Foo
  extend Bar
  extend Baz
  include Qux
  include Norf
  include Grault
  include Thud
end
```

### Style 2

```ruby
class Foo
  extend Bar, Baz
  include Qux, Norf, Grault, Thud
end
```

It would be nice to be able to enforce either of these styles

### PR DESCRIPTION
**Pull request by Drenmi**
This cop checks for grouping of `mixins` in class and `module` bodies. By default, it enforces a separated style, i.e.:

```
class Foo
  include Bar
  include Qux
end
```

but it can be configured to enforce a grouped style as well.

It's currently Chinese New Year, so I didn't have time to add autocorrect. Will add it in at a later point, unless someone beats me to it. 😀 

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] All tests are passing.
* [X] The new code doesn't generate RuboCop offenses.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/cop/style/mixin_grouping.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      #
+      #   @good
+      #   class Foo
+      #     extend Bar, Qox
+      #   end
+      class MixinGrouping < Cop
+        include ConfigurableEnforcedStyle
+
+        MIXIN_METHODS = [:extend, :include, :prepend].freeze
+        MSG = 'Put mixins in %s.'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/style/mixin_grouping.rb]: I think the message is a bit confusing and should definitely mention `include` or `extend` for extra clarity.

File: config/default.yml [hunk 2]
```diff
@@ -728,6 +728,12 @@ Style/MethodName:
     - snake_case
     - camelCase
 
+Style/MixinGrouping:
+  EnforcedStyle: separated
+  SupportedStyles:
```
Comment (bbatsov) [inline config/default.yml]: Some explanations about the supported styles would be useful.

**Comment by bbatsov** [comment]
A couple of points:

* this needs a matching style guide rule
* eventually we need cops for related things like `attr_*` and similar.

**Comment by Drenmi** [comment]
@bbatsov: Comments addressed. 😀 

> this needs a matching style guide rule

Opened a [PR](https://github.com/bbatsov/ruby-style-guide/pull/622) for this.

> eventually we need cops for related things like attr_* and similar.

I was holding off on this one, because I didn't have time to deal with the visibility scopes of the `attr_*` methods, but I'll open a separate issue for it. 😀

**Comment by bbatsov** [comment]
👍

File: manual/cops_style.md [hunk 3]
```diff
@@ -2765,6 +2765,54 @@ EnforcedStyle | both
...
+EnforcedStyle: separated (default)
+
+# bad
+class Foo
+  include Bar, Qox
+end
+
+# good
+class Foo
+  include Bar
```
Comment (PikachuEXE) [inline manual/cops_style.md]:
```ruby
include Bar, Qox
```
==
```ruby
include Qox
include Bar
```

File: manual/cops_style.md [hunk 4]
```diff
@@ -2765,6 +2765,54 @@ EnforcedStyle | both
...
+
+# bad
+class Foo
+  extend Bar
+  extend Qox
+end
+
+# good
+class Foo
+  extend Bar, Qox
```
Comment (PikachuEXE) [inline manual/cops_style.md]:
```ruby
include Bar
include Qox
```
==
```ruby
include Qox, Bar
```

**Pull request merged**